autoload/codi.vim

```diff
@@ -828,3 +828,32 @@ function! codi#complete(arg_lead, cmd_line, cursor_pos)
     endif
     return sort(candidates)
 endfunction
+
+function! codi#new(ft)
+  noswapfile hide enew
+  setlocal buftype=nofile
+  setlocal bufhidden=hide
+
+  call codi#run(0, a:ft)
+endfunction
+
+lua << EOF
+function _G.codi_select(interpreters)
+  local filetypes = {}
+  local n = 0
+  for k, v in pairs(interpreters) do
+    n = n + 1
+    filetypes[n] = k
+  end
+
+  vim.ui.select(filetypes, {
+    prompt = "Codi Filetype",
+  }, function(ft)
+    vim.fn["codi#new"](ft)
+  end)
```

Comment:
My Lua is a bit rusty, but since functions are first-class values, I think we can do the following. Please correct me if I'm wrong, though!

```suggestion
  vim.ui.select(filetypes, {
    prompt = "Codi Filetype",
  }, vim.fn["codi#new"])
```

Reply:
Yes, functions are first-class, but the on_select callback has two arguments (the value and the index of the value). In vimscript, having more arguments than the function is an error, but in lua, extra arguments are ignored. This function is here simply to ignore the extra `index` argument.